com.unity.multiplayer.mlapi/Runtime/Messaging/RpcQueue/RpcQueueContainer.cs

```diff
@@ -571,62 +560,41 @@ public RpcQueueHistoryFrame GetQueueHistoryFrame(RpcQueueHistoryFrame.QueueFrame
-                    {
-                        var rpcQueueHistoryItemInbound = GetQueueHistoryFrame(RpcQueueHistoryFrame.QueueFrameType.Inbound, netUpdateStage);
-                        ResetQueueHistoryFrame(rpcQueueHistoryItemInbound);
-                    }
-
-                    var pooledNetworkBuffer = PooledNetworkBuffer.Get();
-                    var rpcFrameQueueItem = rpcQueueHistoryItemOutbound.GetFirstQueueItem();
-
-                    while (rpcFrameQueueItem.QueueItemType != QueueItemType.None)
-                    {
-                        pooledNetworkBuffer.SetLength(rpcFrameQueueItem.StreamSize);
-                        pooledNetworkBuffer.Position = 0;
-                        byte[] pooledNetworkStreamArray = pooledNetworkBuffer.GetBuffer();
-                        Buffer.BlockCopy(rpcFrameQueueItem.MessageData.Array ?? Array.Empty<byte>(), rpcFrameQueueItem.MessageData.Offset, pooledNetworkStreamArray, 0, (int)rpcFrameQueueItem.StreamSize);
-
-                        if (!IsUsingBatching())
-                        {
-                            pooledNetworkBuffer.Position = 1;
-                        }
+            ProcessAndFlushRpcQueue(RpcQueueProcessingTypes.Receive, updateStage);
 
-                        AddQueueItemToInboundFrame(rpcFrameQueueItem.QueueItemType, UnityEngine.Time.realtimeSinceStartup, rpcFrameQueueItem.NetworkId, pooledNetworkBuffer);
-                        rpcFrameQueueItem = rpcQueueHistoryItemOutbound.GetNextQueueItem();
-                    }
-                }
+            if (updateStage == NetworkUpdateStage.PostLateUpdate)
+            {
+                ProcessAndFlushRpcQueue(RpcQueueProcessingTypes.Send, updateStage);
             }
         }
-#endif
 
         /// <summary>
-        /// Initialize
-        /// This should be called during primary initialization period (typically during NetworkManager's Start method)
         /// This will allocate [maxFrameHistory] + [1 currentFrame] number of PooledNetworkBuffers and keep them open until the session ends
         /// Note: For zero frame history set maxFrameHistory to zero
         /// </summary>
         /// <param name="maxFrameHistory"></param>
-        public void Initialize(uint maxFrameHistory)
+        private void Initialize(uint maxFrameHistory)
         {
+            //This makes sure that we don't exceed a rediculous value by capping the number of queue history frames to ushort.MaxValue
+            //If this value is exceeded, then it will be kept at the ceiling of ushort.Maxvalue.
+            //Note: If running at a 60pps rate (16ms update frequency) this would yield 17.47 minutes worth of queue frame history.
+            if (maxFrameHistory > ushort.MaxValue)
+            {
+                if (NetworkLog.CurrentLogLevel == LogLevel.Developer)
+                {
+                    NetworkLog.LogWarning($"The maximum {nameof(RpcQueueHistoryFrame)} is {ushort.MaxValue}!");
```

Comment:
Could we craft this message to be a bit more actionable, like: "frame history size has exceeded the max of {ushort.MaxValue},".

Also, contrary to the statement, I see that it's logging a warning, but I don't see where it's capping it. Shouldn't there be a `maxFrameHistory = ushort.MaxValue` somewhere?

Reply:
Oh gosh... I will create a new PR for that specific fix.  It should cap the value for sure.